src/server/active_tran_server.cpp

```diff
@@ -208,15 +206,15 @@ active_tran_server::init_page_server_hosts (const char *db_name)
   er_log_debug (ARG_FILE_LINE, "Transaction server runs on %s storage.",
 		m_uses_remote_storage ? "remote" : "local");
 
-  assert (exit_code == NO_ERROR);
-  return exit_code;
+  // failed to connect to any page server
+  assert (valid_connection_count);
+  return valid_connection_count ? NO_ERROR : exit_code;
```

Comment:
According to the table at the start of the function, not having any valid connection, and if remote storage is not specified, is OK.
However, I am not so sure anymore. Therefore, we just return some error as we do. And please update the table.

But, to prevent early out and also to only log successful message when actually successfull:
```
  if (valid_connection_count == 0 && m_uses_remote_storage)
    {
      assert (exit_code != NO_ERROR);
      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_NO_PAGE_SERVER_CONNECTION, 0);
    }
else if (valid_connection_count == 0)
  {
      // failed to connect to any page server
      assert (exit_code != NO_ERROR);
      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, exit_code, 0);
 // NOTE: not sure what exit_code is in this case ...
  }
else
  {
      assert (exit_code == NO_ERROR);
  er_log_debug (ARG_FILE_LINE, "Transaction server runs on %s storage.",
		m_uses_remote_storage ? "remote" : "local");
  }
  
  return exit_code;
```

Note:
regarding
```
assert (valid_connection_count);
```
Don't force an integer into a bool. Be explicit with `> 0`. On the pedantic side, `assert(-1)` will also pass even if this is not what we want from a `count`.

Reply:
done